admin/lint_json.m

```diff
@@ -8,27 +8,35 @@ function lint_json(filesin, outputfile)
 % outputfile           char array of filename to write output to (will overwrite)
 %                          if outputfile is empty will write to stdout
 
-    if ~exist('filesin', 'var') || (exist('filesin', 'var') && isempty(filesin))
-        % Default to glob all
-        filesin = {['**', filesep, '*.m']};
-    end
-    if ~exist('outputfile', 'var') || (exist('outputfile', 'var') && isempty(outputfile))
+    p = inputParser;
+    addOptional(p, 'filesin', {['**', filesep, '*.m']}, @iscellstr);
+    addOptional(p, 'outputfile', '_screen', @ischar);
+    addParameter(p, 'exclude', {}, @iscellstr);
+    parse(p, filesin, outputfile, varargin{:});
+
+    if strcmp(p.Results.outputfile, '_screen')
         fh = 1;
     else % Open file
-        fh = fopen(outputfile,'w');
+        fh = fopen(p.Results.outputfile,'w');
```

Comment:
Using data directly from `p.Results` smears the arg parser though the whole function. It would be cleaner to pull the variables we're interested in out of this structure.